src/hooks/useUser/IntercomProvider.tsx

```diff
@@ -0,0 +1,32 @@
+import Intercom from '@intercom/messenger-js-sdk';
+import { PropsWithChildren, useEffect } from 'react';
+import { User } from './types';
+import env from '@config/env';
+
+type IntercomProviderProps = PropsWithChildren & {
+  user: Partial<User> | null;
+};
+
+export default function IntercomProvider({ children, user }: IntercomProviderProps) {
+  // NOTE: Boot Intercom on app start
+  useEffect(() => {
+    Intercom({ app_id: env.intercomAppId });
+  }, []);
+
+  // NOTE: Set or unset User data on Intercom
+  useEffect(() => {
+    if (!user) {
+      Intercom({ app_id: env.intercomAppId, email: '', name: '', id: '' });
+    } else {
```

Comment:
Snippet for visitors look like this: Just testing
```suggestion
      Intercom({ app_id: env.intercomAppId });
    } else {
```